pulsar-io/core/src/main/java/org/apache/pulsar/io/core/Transformation.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.core;
+
+import org.apache.pulsar.functions.api.Record;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Applies the transformation if the predicate is true.
+ * @param <T>
+ */
+public interface Transformation<T> extends Predicate<Record<T>> {
```

Comment:
would this be `extends Predicate<Record<?>>` instead for the predicate to be useful and consistent?

Reply:
It depends on your Transformation implementation, you can implement a Transformation<Object> to filter out messages with a given message property or any other criteria, without knowing at compile time what is the Record type.